cli/config.py

```diff
@@ -0,0 +1,315 @@
+                'CarbonBlack API token (only needs binary read access): ').strip().lower()
+            if re.fullmatch(self.VALID_CB_API_TOKEN_FORMAT, api_token, re.ASCII):
+                break
+            else:
+                print('ERROR: {}-character input does not match expected token format {}'.format(
+                    len(api_token), self.VALID_CB_API_TOKEN_FORMAT
+                ))
+
+        # We need the KMS key to encrypt the API token.
+        # The same key will be used by the downloader to decrypt the API token at runtime.
+        print('Terraforming KMS key...')
+        os.chdir(TERRAFORM_DIR)
+        subprocess.check_call(['terraform', 'init'])
+        subprocess.check_call(
+            ['terraform', 'apply', '-target=aws_kms_alias.encrypt_credentials_alias']
+        )
+
+        print('Encrypting API token...')
+        response = boto3.client('kms').encrypt(
+            KeyId='alias/{}_binaryalert_carbonblack_credentials'.format(self.name_prefix),
+            Plaintext=api_token
+        )
+        self.encrypted_carbon_black_api_token = base64.b64encode(
+            response['CiphertextBlob']).decode('utf-8')
+
+    def _configure_carbon_black(self) -> None:
+        """If CarbonBlack downloader is enabled, request URL and credentials"""
+        while True:  # CarbonBlack URL
+            try:
+                self.carbon_black_url = get_input('CarbonBlack URL', self.carbon_black_url)
+                break
+            except InvalidConfigError as error:
+                print('ERROR: {}'.format(error))
+
+        update_api_token = 'yes'
+        if self.encrypted_carbon_black_api_token:
+            # API token already exists - ask if they want to update it.
+            while True:
+                update_api_token = get_input('Change the CarbonBlack API token?', 'no')
+                if update_api_token in {'yes', 'no'}:
+                    break
+                else:
+                    print('ERROR: Please enter exactly "yes" or "no"')
+
+        if update_api_token == 'yes':
+            self.save()  # Save updates so far to enable the downloader for terraform.
+            self._encrypt_cb_api_token()
+
+    def configure(self) -> None:
+        """Request basic configuration settings from the user.
+
+        Each request will be retried until the answer is in the correct format.
+        """
+        while True:  # Get AWS account ID
+            try:
+                self.aws_account_id = get_input('AWS Account ID', self.aws_account_id)
+                break
+            except InvalidConfigError as error:
+                print('ERROR: {}'.format(error))
+
+        while True:  # Get AWS region.
+            try:
+                self.aws_region = get_input('AWS Region', self.aws_region)
+                break
+            except InvalidConfigError as error:
+                print('ERROR: {}'.format(error))
+
+        while True:  # Get name prefix.
```

Comment:
Good idea! Yeah, the looping logic always annoyed me. It's a little weird, because there are different use cases and validation paths, but I managed to get the loop into `get_input` like you suggested